rainier-core/src/main/scala/com/stripe/rainier/compute/Target.scala

```diff
@@ -18,6 +18,45 @@ class Target(val real: Real, val placeholders: Map[Variable, Array[Double]]) {
 
   private def nRows(data: Map[Variable, Array[Double]]): Int =
     data.head._2.size
+
+  def batched(nBatches: Int): (Target, Real) =
+    if (placeholders.isEmpty)
+      (this, Real.zero)
+    else if (nBatches == 1)
+      (Target.empty, inlined)
+    else {
+      val n = nRows(placeholders)
+      val variables = placeholders.keys.toList
+      val nColumns = n / nBatches
+      val copies = 0.until(nColumns).map { i =>
```

Comment:
Since `batched` already has dense logic, even though the code is not particularly dense, it might still help to move the `copies, newPlaceholders` and `remainder` logic into private helper functions so the body of `batched` itself is super streamlined.

Reply:
ok sounds good